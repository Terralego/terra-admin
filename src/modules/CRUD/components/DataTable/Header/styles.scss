.table-header {
  display: flex;
  flex-wrap: wrap;
  padding: 0.2rem 0;
  align-items: center;
  justify-content: space-between;
  
  &__title {
    display: flex;
    align-items: center;
    strong {
      margin-left: 0.2rem;
    }
  }

  &__spinner {
    margin-right: 0.2rem;
  }

  &__create {
    display: none;
    margin-left: 0.5rem;
  }

  &__actions {
    display: flex;
    > * {
      margin: 0.2rem;
    }
  }

  &__resize {
    > button {
      margin: 0 0.1rem;
    }
  }
}

@media all and (min-width: 42em) {
  .table-header {
    &__create {
      display: block;
    }
  }
}
