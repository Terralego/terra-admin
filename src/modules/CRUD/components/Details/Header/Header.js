import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import FitBoundButton from '../FitBoundButton';
import ExportFiles from '../ExportFiles';

const Header = ({ className, generatedFiles, geomFiles, title, ...props }) => (
  <header className={classnames('details__header', className)} {...props}>
    <div className="details__header-title">
      <FitBoundButton title={title} />
      <h2 className="details__title">{title}</h2>
    </div>
    <ExportFiles generatedFiles={generatedFiles} geomFiles={geomFiles} />
  </header>
);

Header.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string.isRequired,
  generatedFiles: PropTypes.arrayOf(
    PropTypes.shape({
      template_name: PropTypes.string,
      download_url: PropTypes.string,
      template_file: PropTypes.string,
    }),
  ),
};

Header.defaultProps = {
  className: '',
  generatedFiles: undefined,
};

export default Header;
