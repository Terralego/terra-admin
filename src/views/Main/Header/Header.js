import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import {
  Navbar,
  NavbarGroup,
  NavbarHeading,
  Classes,
} from '@blueprintjs/core';

import UserDropdown from '../UserDropdown';
import MenuDropdown from '../MenuDropdown';
// import LangDropdown from '../LangDropdown';

import './header.scss';

const Logo = ({
  t,
  title = t('main.title'),
  logo: {
    src = undefined,
    alt = title,
  },
}) => {
  if (!src) {
    return <span>{alt}</span>;
  }
  return <img className="main-header__logo" src={src} alt={alt} />;
};

export const Header = ({ authenticated, heading, ...props }) => (
  <header className="main-header">
    <Navbar className={Classes.DARK}>
      <NavbarGroup align="left">
        <NavbarHeading>
          <NavLink to="/">
            <Logo {...props} />
          </NavLink>
        </NavbarHeading>
      </NavbarGroup>
      {heading && (
        <NavbarGroup>
          { /* eslint-disable-next-line react/no-danger */ }
          <div dangerouslySetInnerHTML={{ __html: heading }} />
        </NavbarGroup>
      )}
      {authenticated && (
        <NavbarGroup align="right">
          <MenuDropdown />
          <UserDropdown />
        </NavbarGroup>
      )}
      {/* Deactivation of langDropDown
      <NavbarGroup align="right">
        <LangDropdown />
      </NavbarGroup>
      */}
    </Navbar>
  </header>
);

Header.propTypes = {
  t: PropTypes.func,
  title: PropTypes.string,
  logo: PropTypes.shape({
    src: PropTypes.string,
    alt: PropTypes.string,
  }),
};

Header.defaultProps = {
  t: text => text,
  title: undefined,
  logo: {
    src: undefined,
    alt: undefined,
  },
};

export default Header;
