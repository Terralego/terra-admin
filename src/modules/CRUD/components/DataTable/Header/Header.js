import React from 'react';
import PropTypes from 'prop-types';

import {
  Icon,
  Spinner,
} from '@blueprintjs/core';
import { NavLink } from 'react-router-dom';

import ExportGeom from '../ExportGeom';
import { generateURI } from '../../../config';
import Actions from '../Actions';

import './styles.scss';

const Header = ({
  columns,
  displayAddFeature,
  featuresList: { count } = {},
  match: { params: { layer } },
  onChange,
  onHeaderChange,
  layerName,
  t,
}) => (
  <div className="table-header">
    <div className="table-header__title">
      {count === undefined && <Spinner className="table-header__spinner" size={Spinner.SIZE_SMALL} tagName="span" />}
      <span>{t('CRUD.table.results', { count })}</span>
      <strong>{layerName}</strong>
      {displayAddFeature && (
        <NavLink className="table-header__create" to={generateURI('layer', { layer, id: 'create' })}>
          <span className="bp3-button bp3-minimal">
            <Icon icon="add" />
            <span className="bp3-button-text"> {t('CRUD.details.create')}</span>
          </span>
        </NavLink>
      )}
      <ExportGeom />
    </div>
    <Actions
      columns={columns}
      onHeaderChange={onHeaderChange}
      onChange={onChange}
    />
  </div>
);

Header.propTypes = {
  layerName: PropTypes.string,
  t: PropTypes.func,
  match: PropTypes.shape({
    params: PropTypes.shape({
      layer: PropTypes.string,
    }),
  }),
  displayAddFeature: PropTypes.bool,
  featuresList: PropTypes.shape({
    count: PropTypes.number,
  }),
};

Header.defaultProps = {
  layerName: '',
  t:  () => {},
  match: {
    params: {
      layer: undefined,
    },
  },
  displayAddFeature: false,
  featuresList: {
    count: undefined,
  },
};

export default Header;
