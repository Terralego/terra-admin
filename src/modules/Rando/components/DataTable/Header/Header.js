import React from 'react';
import { Intent, Position, Popover, PopoverInteractionKind, Button, Icon } from '@blueprintjs/core';
import ColumnsSelector from '@terralego/core/modules/Table/components/ColumnsSelector';
import { NavLink } from 'react-router-dom';
import { generateURI } from '../../../config';

import './styles.scss';

const Header = ({ source, full, resize, t, columns, onChange, match: { params: { layer } } }) => (
  <div className="table-header">
    <div className="table-header__title">
      {t('rando.table.title', { source })}
      <NavLink className="table-header__create" to={generateURI('layer', { layer, action: 'create' })}>
        <span className="bp3-button">
          <Icon icon="plus" />
          <span className="bp3-button-text"> {t('rando.details.create')}</span>
        </span>
      </NavLink>
    </div>
    <div>
      <Popover
        content={t('rando.table.filterProps')}
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <ColumnsSelector
          columns={columns}
          onChange={onChange}
          position={Position.LEFT}
          locales={{
            displayAllColumns: t('rando.table.columnsDisplay'),
            hideAllColumns: t('rando.table.columnsHide'),
          }}
        />
      </Popover>
      <Popover
        content={full ? t('rando.table.minimize') : t('rando.table.maximize')}
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <Button
          onClick={() => resize(!full)}
          icon={full ? 'minimize' : 'maximize'}
          minimal
          active={full}
          intent={Intent.PRIMARY}
        />
      </Popover>
    </div>
  </div>
);

export default Header;
