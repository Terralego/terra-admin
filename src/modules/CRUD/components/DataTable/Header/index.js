import { withRouter } from 'react-router-dom';
import { withTranslation } from 'react-i18next';
import { connectAuthProvider } from '@terralego/core/modules/Auth';
import { connectCRUDProvider } from '../../../services/CRUDProvider';
import { withTableSize } from '../../../services/UserSettingsProvider';
import compose from '../../../../../utils/compose';

import Header from './Header';

const authProviderGetter = ({
  authenticated,
  user,
}) => {
  const permissions = authenticated ? user.permissions : [];
  return {
    displayAddFeature: permissions.includes('can_add_feature'),
  };
};

export default compose(
  withRouter,
  connectAuthProvider(authProviderGetter),
  connectCRUDProvider('featuresList'),
  withTableSize(),
  withTranslation(),
)(Header);
