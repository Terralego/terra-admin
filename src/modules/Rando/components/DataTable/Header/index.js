import { withRouter } from 'react-router';
import { withNamespaces } from 'react-i18next';
import { connectAuthProvider } from '@terralego/core/modules/Auth';

import Header from './Header';

export default connectAuthProvider(({
  authenticated,
  user,
}) => {
  const permissions = authenticated ? user.permissions : [];
  return {
    displayAddFeature: permissions.includes('terra.add_feature'),
  };
})(
  withNamespaces()(
    withRouter(Header),
  ),
);
