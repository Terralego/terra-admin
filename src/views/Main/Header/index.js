import { connectAuthProvider } from '@terralego/core/modules/Auth';
import { withTranslation } from 'react-i18next';
import { connectAppProvider } from '../../../components/AppProvider';

import Header from './Header';

export default connectAppProvider(({
  env: {
    title,
    theme: { logo, heading } = {},
  },
}) => ({
  title,
  logo,
  heading,
}))(connectAuthProvider('authenticated')(withTranslation()(Header)));
