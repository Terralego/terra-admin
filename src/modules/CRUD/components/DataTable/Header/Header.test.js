import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

jest.mock('@blueprintjs/core', () => ({
  Icon: ({ children, ...props }) => <span {...props}>{children}</span>,
  Spinner: ({ children, ...props }) => <div {...props}>{children}</div>,
}));

jest.mock('react-router-dom', () => ({
  NavLink: ({ children }) => <div>{children}</div>,
}));

jest.mock('../Actions', () => props => <div {...props} />);
jest.mock('../ExportGeom', () => props => <div {...props} />);

let props;
beforeEach(() => {
  props = {
    layerName: 'Layer Test',
    tableSize: 'medium',
    t:  text => text,
    columns: [],
    onChange:  () => {},
    onHeaderChange:  () => {},
    match: {
      params: {
        layer: 'layer',
      },
    },
    displayAddFeature: false,
    featuresList: {},
  };
});

it('should render correctly', () => {
  const tree = renderer.create((
    <Header
      {...props}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
