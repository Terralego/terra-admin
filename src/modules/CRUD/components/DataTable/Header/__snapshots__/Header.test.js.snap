// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="table-header"
>
  <div
    className="table-header__title"
  >
    <div
      className="table-header__spinner"
      tagName="span"
    />
    <span>
      CRUD.table.results
    </span>
    <strong>
      Layer Test
    </strong>
    <div />
  </div>
  <div
    columns={Array []}
    onChange={[Function]}
    onHeaderChange={[Function]}
  />
</div>
`;
